app/controllers/grouptrainingplans_controller.rb

```diff
@@ -0,0 +1,96 @@
+class GrouptrainingplansController < ApplicationController
+  before_action :set_grouptrainingplan, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @grouptrainingplans = Grouptrainingplan.all
+  end
+
+  def show
+  end
+
+  def new
+    @grouptrainingplan = Grouptrainingplan.new
+  end
+
+  def edit
+  end
+
+  def create
+    @grouptrainingplan = Grouptrainingplan.new(grouptrainingplan_params)
+    respond_to do |format|
+      if @grouptrainingplan.save
+        format.html { redirect_to @grouptrainingplan, notice: 'Group Training Plan was successfully create. ' }
+        format.json { render :show, status: :created, location: @grouptrainingplan }
+      else
+        format.html{ render :new }
+        format.json { render json: @grouptrainingplan.errors, status: :unprocessable_entity }
+      end
+    end
+
+    #Will give all runners in given group this training program as individual training programs
+    Runner.find_each do |running|
+      if running.group_id.to_s == grouptrainingplan_params[:group_id]
```

Comment:
Instead of using the params objects, since you went through all that effort to create a real grouptrainingplan object, why not refer to that?    then you can do running.group_id = groputrainingplan.     Heck, you might even be able to do stuff like indPlan.grouptrainingpan = grouptrainingplan.